packages/kbn-generate-csv/src/generate_csv.test.ts

```diff
@@ -427,6 +430,195 @@ describe('CsvGenerator', () => {
+      const timeFromNowInMs = 4 * 60 * 1000;
+
+      const taskInstanceFields = getTaskInstanceFields({
+        seconds: timeFromNowInMs / 1000,
+      });
+
+      mockDataClientSearchFn.mockImplementation((_, options) => {
+        const getSearchResult = () => {
+          const queuedAt = Date.now();
+
+          return new Promise<IKibanaSearchResponse<ReturnType<typeof getMockRawResponse>>>(
+            (resolve, reject) => {
+              setTimeout(() => {
+                if (
+                  new Date(Date.now()).getTime() - new Date(queuedAt).getTime() >
+                  Number((options?.transport?.requestTimeout! as string).replace(/ms/, ''))
+                ) {
+                  reject(
+                    new esErrors.ResponseError({ statusCode: 408, meta: {} as any, warnings: [] })
+                  );
+                } else {
+                  resolve({
+                    rawResponse: getMockRawResponse(
+                      [
+                        {
+                          fields: { a: ['a1'], b: ['b1'] },
+                        } as unknown as estypes.SearchHit,
+                      ],
+                      3
+                    ),
+                  });
+                }
+              }, timeFromNowInMs / 4);
+            }
+          );
+        };
+
+        return Rx.defer(getSearchResult);
+      });
+
+      const generateCsvPromise = new CsvGenerator(
+        createMockJob({ searchSource: {}, columns: ['a', 'b'] }),
+        mockConfigWithAutoScrollDuration,
+        taskInstanceFields,
+        {
+          es: mockEsClient,
+          data: {
+            ...mockDataClient,
+            search: mockDataClientSearchFn,
+          },
+          uiSettings: uiSettingsClient,
+        },
+        {
+          searchSourceStart: mockSearchSourceService,
+          fieldFormatsRegistry: mockFieldFormatsRegistry,
+        },
+        new CancellationToken(),
+        mockLogger,
+        stream
+      ).generateData();
+
+      await jest.advanceTimersByTimeAsync(timeFromNowInMs);
+
+      expect(await generateCsvPromise).toEqual(
+        expect.objectContaining({
+          warnings: [],
+        })
+      );
+
+      expect(mockDataClientSearchFn).toBeCalledWith(
+        { params: { body: {}, ignore_throttled: undefined, max_concurrent_shard_requests: 5 } },
+        { strategy: 'es', transport: { maxRetries: 0, requestTimeout: `${timeFromNowInMs}ms` } }
+      );
+
+      expect(content).toMatchSnapshot();
+    });
+
+    it('csv generation errors if search request does not resolve before the computed timeout value passed to the search data client elapses', async () => {
+      const timeFromNowInMs = 4 * 60 * 1000;
+
+      const taskInstanceFields = getTaskInstanceFields({
+        seconds: timeFromNowInMs / 1000,
+      });
+
+      const requestDuration = timeFromNowInMs + 1000;
```

Comment:
Configure request duration that exceeds the timeout value passed to search client